sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/implementation/PhoneNumberAdminClientImpl.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.communication.administration.implementation;
+
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+
+/** Initializes a new instance of the PhoneNumberAdminClient type. */
+public final class PhoneNumberAdminClientImpl {
+    /** The endpoint of the Azure Communication resource. */
+    private final String endpoint;
+
+    /**
+     * Gets The endpoint of the Azure Communication resource.
+     *
+     * @return the endpoint value.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+    /** Api Version. */
+    private final String apiVersion;
+
+    /**
+     * Gets Api Version.
+     *
+     * @return the apiVersion value.
+     */
+    public String getApiVersion() {
+        return this.apiVersion;
+    }
+
+    /** The HTTP pipeline to send requests through. */
+    private final HttpPipeline httpPipeline;
+
+    /**
+     * Gets The HTTP pipeline to send requests through.
+     *
+     * @return the httpPipeline value.
+     */
+    public HttpPipeline getHttpPipeline() {
+        return this.httpPipeline;
+    }
+
+    /** The PhoneNumberAdministrationsImpl object to access its operations. */
+    private final PhoneNumberAdministrationsImpl phoneNumberAdministrations;
+
+    /**
+     * Gets the PhoneNumberAdministrationsImpl object to access its operations.
+     *
+     * @return the PhoneNumberAdministrationsImpl object.
+     */
+    public PhoneNumberAdministrationsImpl getPhoneNumberAdministrations() {
+        return this.phoneNumberAdministrations;
+    }
+
+    /** Initializes an instance of PhoneNumberAdminClient client. */
+    PhoneNumberAdminClientImpl(String endpoint) {
+        this(
+                new HttpPipelineBuilder()
+                        .policies(new UserAgentPolicy(), new RetryPolicy(), new CookiePolicy())
+                        .build(),
+                endpoint);
+    }
+
+    /**
+     * Initializes an instance of PhoneNumberAdminClient client.
+     *
+     * @param httpPipeline The HTTP pipeline to send requests through.
+     */
+    PhoneNumberAdminClientImpl(HttpPipeline httpPipeline, String endpoint) {
+        this.httpPipeline = httpPipeline;
+        this.endpoint = endpoint;
+        this.apiVersion = "2020-07-20-preview1";
```

Comment:
Shouldn't these get picked from some configuration rather than being hardcoded?

Reply:
This code is generated by AutoRest.